contrib/omhttp/omhttp.c

```diff
@@ -1772,6 +1816,14 @@ CODESTARTnewActInst
 			pData->httpheaderkey = (uchar*)es_str2cstr(pvals[i].val.d.estr, NULL);
 		} else if(!strcmp(actpblk.descr[i].name, "httpheadervalue")) {
 			pData->httpheadervalue = (uchar*)es_str2cstr(pvals[i].val.d.estr, NULL);
+		} else if(!strcmp(actpblk.descr[i].name, "httpheaders")) {
+			pData->nHttpHeaders = pvals[i].val.d.ar->nmemb;
+			CHKmalloc(pData->httpHeaders = malloc(sizeof(uchar *) * pvals[i].val.d.ar->nmemb ));
+			for(int j = 0 ; j <  pvals[i].val.d.ar->nmemb ; ++j) {
+				char *cstr = es_str2cstr(pvals[i].val.d.ar->arr[j], NULL);
+				CHKiRet(processHeaderParam(cstr, (const uchar **)&pData->httpHeaders[j]));
```

Comment:
Since you already own the memory to `cstr` here, is it necessary to do a `strdup` inside `processHeaderParam` and then free the string in the next line? Can this function just do validation?

Reply:
Got it, doing some adjustments